### ISSUE TITLE
Improvements to SvarDOS Packages page

### ISSUE THREAD

**Issue by bttrx** [SvarDOS/bugz#52]
(pasting from http://svardos.org/?p=forum&thread=1705448028)

Talking about http://www.svardos.org/?p=repo

1. Would be nice to see the category of each package while scrolling the list. (But I see, we currently have a few 'special' packages, e.g., 'command', 'dosfsck', and 'fdmore', belonging to multiple categories.)
2. Maybe also make that 'category' cell clickable to apply a filter to the list, e.g., I see 'ivan' is in category 'games' and can click on 'games' next to 'ivan' to see all games in the repo.
3. Maybe apply the target hardware as tags to each package.

By the way: Maybe we should rename 'progs' to 'apps' these days? (Not only for the Packages page, but also in SvarDOS' directory structure.

**Comment by bttrx** [comment]
You didn't ask for, but more to come:

4. Show package size (in Kbytes?)
5. Advanced (coz this data is not present in most LSM files): Show license type, e.g., 'BSD', 'MIT', 'GPLv2, 'GPLv3', 'proprietary (= ask your favorite lawyer)', ...
6. Provide a link to the source code package if any.
7. Provide a search button instead of using a browser's search function

**Comment by bttrx** [comment]
8. https://github.com/SvarDOS/bugz/issues/51#issuecomment-1897957701

**Comment by igully** [comment]
The big problem I see is that it would require more manpower to actively and continuously classify each package, gather its information, etc. I don´t know whether such manpower exists to budget that increase on workload.

However, it is still a good idea.

**Comment by bttrx** [comment]
> would require more manpower

Yes, of course. Maybe (wo)manpower comes after the functions have been made available.

> However, it is still a good idea.

Glad you like it.

**Comment by mateuszviste** [comment]
Seems to me that a full-time php developer would be needed here :-)

All this makes sense of course, not sure I will be able to address it all any time soon, though. requires reworking of how it is all displayed on the repo page to not overload the screen estate.

About displaying (gathering/classifying/etc) the license of each package: I'm not sure there is much added value to it. I mean the information would be potentially a "nice to have" thing, but the necessary manpower to do it properly is big, while in practice I somehow doubt that the license of a package is a criterion for someone to decide of installing or not a given package. At least I know that I do not care about licensing, as long as it's legal and functional. This is probably something important to open-source purists, GNU zealots and some such other leftists, but given the SvarDOS stance on accepting packages I doubt that this population is ever to be part of our user base :)
(one of the reasons I started SvarDOS actually: I was really annoyed by all the licensing-drama around and within FreeDOS - as long as it works and nobody's harmed I'm fine it)

**Comment by bttrx** [comment]
I mentioned licenses only for completeness, because some visitors might care about more than you and me.

**Comment by mateuszviste** [comment]
proof of concept: http://svardos.org/?p=repo&view=1
for now only category is added there (non clickable), as well as a new LSM field named 'url'.
